pycalphad/tests/fixtures.py

```diff
@@ -26,3 +28,12 @@ def test_name_of_my_test(load_database):
     ```
     """
     return pytest.mark.parametrize("load_database", [path], indirect=True)
+
+
+class ConvergenceFailureSolver(Solver):
+    """Solver that is guaranteed to produce a convergence failure for any composition sets or conditions passed"""
+    def solve(self, composition_sets, conditions):
+        spec = self.get_system_spec(composition_sets, conditions)
+        self._fix_state_variables_in_compsets(composition_sets, conditions)
+        state = spec.get_new_state(composition_sets)
+        return SolverResult(converged=False, x=composition_sets[0].dof, chemical_potentials=np.full_like(state.chemical_potentials, np.nan))
```

Comment:
So we are only setting NaN in chemical potentials, but otherwise returning the minimizer solution? This also looks like it would break calling code if more than one composition set is passed in as input.

Reply:
I think it should't matter what the chemical potentials are set to. In real non-converging cases, the chemical potentials will not be NaN and the value of `x` should also not matter (unless there are cases where we actually expect to be able to produce any meaningful result from a convergence failure, but right now I don't think that we do).

Reply:
I set them to a nonsense value instead of NaN in the `SolverResult`, as that's more representative of realistic cases.